pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -190,7 +192,13 @@ def compute_decomposition(
 
         single_round = []
         with qml.QueuingManager.stop_recording():
-            for pw, coeff in hamiltonian.pauli_rep.items():
+            try:
+                pauli_rep = qml.pauli.pauli_sentence(hamiltonian)
```

Comment:
I didnt know this works!
So adding pauli rep to `Hamiltonian` would be a matter of 
```
def _build_pauli_rep(self):
    return qml.pauli.pauli_sentence(self)
```
to `Hamiltonian` class

Reply:
With a try-except that falls back returns `None` if there's an error.  

So yes, I think it would actually be worth adding a couple of lines to `Hamiltonian` just to make things like this easier.

Reply:
PR #5377  . I'll update this PR once that's merged in.